src/Fmt_ast.ml

```diff
@@ -3639,14 +3639,7 @@ and fmt_structure c ctx itms =
           | _, Pstr_include {pincl_mod= me} | _, Pstr_module {pmb_expr= me} ->
               is_simple_mod me
           | _, Pstr_open _ -> true
-          | `Compact, _ ->
-              let width i =
-                let last = false in
-                String.length
-                  (Format.asprintf "%t"
-                     (Cmts.preserve (fun () -> fmt_structure_item ~last c i)))
```

Comment:
The argument function to preserve does not get called until asprintf runs and passes it the formatting stream. At that time preserve has already reset the remove flag, and so preserve doesn't work. Sorry, this is an existing but unknown bug in preserve. I'm looking into a fix.